common/rpc/rpc.go

```diff
@@ -67,18 +79,24 @@ func NewFactory(
 	logger log.Logger,
 	tlsProvider encryption.TLSConfigProvider,
 	frontendURL string,
+	frontendHTTPURL string,
 	frontendTLSConfig *tls.Config,
 	clientInterceptors []grpc.UnaryClientInterceptor,
+	monitor membership.Monitor,
 ) *RPCFactory {
-	return &RPCFactory{
+	f := &RPCFactory{
 		config:             cfg,
 		serviceName:        sName,
 		logger:             logger,
 		frontendURL:        frontendURL,
+		frontendHTTPURL:    frontendHTTPURL,
 		frontendTLSConfig:  frontendTLSConfig,
 		tlsFactory:         tlsProvider,
 		clientInterceptors: clientInterceptors,
+		monitor:            monitor,
 	}
+	f.localFrontendClient = sync.OnceValues(f.createLocalFrontendHTTPClient)
```

Comment:
I have a slight preference to use the same `sync.Once` mechanism as we're using for `initListener` and `grpcListener` for the local frontend client.

Reply:
Why? This is a newer mechanism that's perfect for this use case.

Reply:
For consistency, mostly. Not a big deal though.